libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java

```diff
@@ -18,126 +21,100 @@
 import java.io.InputStreamReader;
 import java.lang.reflect.Type;
 import java.util.ArrayList;
-import java.util.Date;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 
 import okhttp3.Call;
 import okhttp3.Callback;
-import okhttp3.HttpUrl;
 import okhttp3.OkHttpClient;
-import okhttp3.Request;
 import okhttp3.Response;
+import okhttp3.ResponseBody;
 
 class CertificateBlacklist implements Callback {
   private static final String LOG_TAG = "MapboxBlacklist";
   private static final String BLACKLIST_FILE = "MapboxBlacklist";
   private static final String SHA256 = "sha256/";
-  private static final String NEW_LINE = "\n";
-  private static final String HTTPS = "https://";
-  private static final String BACKSLASH = "/";
-  private static final String EMPTY_STRING = "";
-  private static final int BLACKLIST_HEAD = 0;
   private static final long DAY_IN_MILLIS = 86400000;
-  private static final String COM_CONFIG_ENDPOINT = "api.mapbox.com";
-  private static final String CHINA_CONFIG_ENDPOINT = "api.mapbox.cn";
-  private static final String ACCESS_TOKEN_QUERY_PARAMETER = "access_token";
   private static final String REQUEST_FAIL = "Request failed to download blacklist";
-  private static final String HTTPS_SCHEME = "https";
-  private static final String USER_AGENT_REQUEST_HEADER = "User-Agent";
-  private static final String EVENT_CONFIG_SEGMENT = "events-config";
-  private static final Map<Environment, String> ENDPOINTS = new HashMap<Environment, String>() {
-    {
-      put(Environment.COM, COM_CONFIG_ENDPOINT);
-      put(Environment.STAGING, COM_CONFIG_ENDPOINT);
-      put(Environment.CHINA, CHINA_CONFIG_ENDPOINT);
-    }
-  };
+  static final String MAPBOX_SHARED_PREFERENCE_KEY_BLACKLIST_TIMESTAMP = "mapboxBlacklistTimestamp";
   private final Context context;
   private final String accessToken;
-  private final String userAgent;
-  private final OkHttpClient client;
+  private  final BlacklistClient blacklistClient;
+  private List<String> revokedKeys;
 
-  CertificateBlacklist(Context context, String accessToken, String userAgent, OkHttpClient client) {
+  CertificateBlacklist(Context context, String accessToken, String userAgent) {
     this.context = context;
     this.accessToken = accessToken;
-    this.userAgent = userAgent;
-    this.client = client;
+    this.blacklistClient = new BlacklistClient(userAgent, new OkHttpClient(), this);
+    retrieveBlackList();
   }
 
-  List<String> retrieveBlackList() {
+  boolean isBlacklisted(String hash) {
+    for (String revokedKey : revokedKeys) {
```

Comment:
`List<T>` has `contains()` method, should we just use it?

Reply:
Due to there being only a single instance of `CERTIFICATE_PINS` within the `CertificatePinnerFactory` class, we can not use a simple `contains()` method. Our Certificates need to keep their `shar256/` prefix, because being added after the `isBlacklisted` check will cause this prefix to be added on each additional request.  (i.e. sha256/sha256/sha256/...) 

We thus have a system where the local hashes have `sha256/` while the save blacklisted ones will not have this prefix. (since it was requested to be removed above). The current solution works within these parameters. An alternative may be switching around the analysis and seeing if any of the blacklisted hashes match the current hashes, but this would go back to our system to before this PR where we are returning a list of hashes.

Reply:
aren't we expecting `isBlacklisted` method to be called just once and not with every network request?